src/core/AutoRest.Core/CodeGenerator.cs

```diff
@@ -139,7 +140,7 @@ public virtual void PopulateSettings(IDictionary<string, object> settings)
                     {
                         await textWriter.WriteLineAsync();
                     }
-                    else if (!string.IsNullOrWhiteSpace(line))
+                    else if (!skipEmptyLines || (skipEmptyLines && !string.IsNullOrWhiteSpace(line)))
```

Comment:
The second check for skipEmptyLines is unnecessary since it can only be true if the second expression is evaluated.